On order Library and in access facet

lmcglohon opened the pull request.

Commit:
Add original properties file

Commit:
Finish implementation

ndushay commented on stanford-sw/translation_maps/library_on_order_map_orig.properties:
```diff
@@ -0,0 +1,34 @@
+# Map ON-ORDER Library Codes to User Friendly Description - Long Version for Facet Values
```
do we really need to keep both?  Old one is in git.  If we do need both, say something in this comment?

lmcglohon replied:
The original file came from Jvine so I thought we needed to keep it in case there was a question later.

ndushay replied:
so neither file existed before?  @jvine:  any reason to keep both versions in the code?   I *think* it's just a matter of how the locations percolate through indexing and UI code - when they get translated ... is that true, Laney?

jvine replied:
i have no opinion.

lmcglohon replied:
It is fine to delete the old file.  Just trying to be thorough.

ndushay commented on stanford-sw/src/edu/stanford/enumValues/Access.java:
```diff
@@ -1,12 +1,14 @@
 	ONLINE,
-	AT_LIBRARY;
+	AT_LIBRARY,
+  ON_ORDER;
```
whitespace alignment off here.  stupid whitespace.

ndushay commented on stanford-sw/src/edu/stanford/enumValues/Access.java:
```diff
@@ -18,10 +20,11 @@ public String toString() {
 		case ONLINE:
 			return "Online";
+		case ON_ORDER:
+			return "On order";
```
I assume @jvine said it's "On order" and not "On Order" ...

jvine replied:
yes, jvine did.

ndushay commented on stanford-sw/test/src/edu/stanford/ItemOnOrderTests.java:
```diff
@@ -0,0 +1,120 @@
+    df999hloc.addSubfield(factory.newSubfield('k', "ON-ORDER"));
+    df999hloc.addSubfield(factory.newSubfield('m', "GREEN"));
+  }
+  DataField df999hcloc = factory.newDataField("999", ' ', ' ');
```
I'm not sure I would both with changes now, but I would suggest that since each DataField is used exactly once for a single test below that this test data would be inline with the actual test. (I would leave factory and ldr up here, tho).

ndushay commented on stanford-sw/test/src/edu/stanford/ItemOnOrderTests.java:
```diff
@@ -0,0 +1,120 @@
+    solrFldMapTest.assertSolrFldValue(record, LibfldName, "Green");
+    solrFldMapTest.assertSolrFldValue(record, AccfldName, "On order");
+
+    // 2. 999 with ON-ORDER in 999$k, library in 999$m
```
Again, not sure it's worth the trouble right now, but I think it would be better to split these into 3 separate test methods.  Is there any reason to group them like this?

Also, might be good to have one library other than Green for the sake of ensuring proper translations (something with a less obvious translation, perhaps?)

ndushay commented on stanford-sw/test/src/edu/stanford/ItemOnOrderTests.java:
```diff
@@ -0,0 +1,120 @@
+    solrFldMapTest.assertNoSolrFld(record, LibfldName);
+    solrFldMapTest.assertSolrFldValue(record, AccfldName, "On order");
+
+    // 5. no 999 but a 596$a with a value in the lookup table
```
ditto to above - this *could* be two separate test methods.  But whatever.

ndushay commented:
Remaining:
* whitespace issue
* do we really want both versions of properties files?  if so, can the comment at the top indicate why we have _orig?
* test refactor:  s'ok to leave it;  wonder if you agree w me for next time.

ndushay commented:
One last thought: again, I'm late to the party, but do we want to ensure that anything that is access "On order"  is not also "At the Library" unless another copy is already there?   Or is that a stupid worry?

lmcglohon commented:
We can fix the tests now if you'd like - or postpone them is fine as well.

lmcglohon commented:
@ndushay I believe I have answered your summary questions:
whitespace issue - done
do we really want both versions of properties files? if so, can the comment at the top indicate why we have _orig?  I removed the original version
test refactor: s'ok to leave it; wonder if you agree w me for next time. - refactored tests
